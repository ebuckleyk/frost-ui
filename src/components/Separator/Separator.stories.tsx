import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';

import { Card } from '../Card';
import { Separator } from './Separator';

function SeparatorDemo() {
  return (
    <Card className="p-5">
      <div className="space-y-1">
        <h4 className="text-sm font-medium leading-none">Radix Primitives</h4>
        <p className="text-sm text-muted-foreground">An open-source UI component library.</p>
      </div>
      <Separator className="my-4" />
      <div className="flex h-5 items-center space-x-4 text-sm">
        <div>Blog</div>
        <Separator orientation="vertical" />
        <div>Docs</div>
        <Separator orientation="vertical" />
        <div>Source</div>
      </div>
    </Card>
  );
}

type ComponentType = React.ComponentProps<typeof Separator>;
const meta: Meta<ComponentType> = {
  component: Separator,
  render: SeparatorDemo,
};

export default meta;

type Story = StoryObj<ComponentType>;
export const Demo: Story = {};
