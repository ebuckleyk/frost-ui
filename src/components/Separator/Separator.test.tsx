import * as React from 'react';
import { render } from '@testing-library/react';

import { Separator } from './Separator';

function Component() {
  return (
    <>
      <div className="space-y-1">
        <h4 className="text-sm font-medium leading-none">Radix Primitives</h4>
        <p className="text-sm text-muted-foreground">An open-source UI component library.</p>
      </div>
      <Separator className="my-4" />
      <div className="flex h-5 items-center space-x-4 text-sm">
        <div>Blog</div>
        <Separator orientation="vertical" />
        <div>Docs</div>
        <Separator orientation="vertical" />
        <div>Source</div>
      </div>
    </>
  );
}

describe('Separator', () => {
  it('should render Separator component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
