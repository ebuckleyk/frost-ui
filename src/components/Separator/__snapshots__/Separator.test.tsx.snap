// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Separator > should render Separator component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="space-y-1"
      >
        <h4
          class="text-sm font-medium leading-none"
        >
          Radix Primitives
        </h4>
        <p
          class="text-sm text-muted-foreground"
        >
          An open-source UI component library.
        </p>
      </div>
      <div
        class="shrink-0 bg-border h-[1px] w-full my-4"
        data-orientation="horizontal"
        role="none"
      />
      <div
        class="flex h-5 items-center space-x-4 text-sm"
      >
        <div>
          Blog
        </div>
        <div
          class="shrink-0 bg-border h-full w-[1px]"
          data-orientation="vertical"
          role="none"
        />
        <div>
          Docs
        </div>
        <div
          class="shrink-0 bg-border h-full w-[1px]"
          data-orientation="vertical"
          role="none"
        />
        <div>
          Source
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="space-y-1"
    >
      <h4
        class="text-sm font-medium leading-none"
      >
        Radix Primitives
      </h4>
      <p
        class="text-sm text-muted-foreground"
      >
        An open-source UI component library.
      </p>
    </div>
    <div
      class="shrink-0 bg-border h-[1px] w-full my-4"
      data-orientation="horizontal"
      role="none"
    />
    <div
      class="flex h-5 items-center space-x-4 text-sm"
    >
      <div>
        Blog
      </div>
      <div
        class="shrink-0 bg-border h-full w-[1px]"
        data-orientation="vertical"
        role="none"
      />
      <div>
        Docs
      </div>
      <div
        class="shrink-0 bg-border h-full w-[1px]"
        data-orientation="vertical"
        role="none"
      />
      <div>
        Source
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
